starter-code/src/app/components/log-form/log-form.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, OnInit } from '@angular/core';
+import { AccessControlLogService } from '../../services/access-control-log.service';
+
+@Component({
+  selector: 'app-log-form',
+  templateUrl: './log-form.component.html',
+  styleUrls: ['./log-form.component.css']
+})
+export class LogFormComponent implements OnInit {
+  variable: any;
+  feedbackEnabled = false;
+  error = null;
+  processing = false;
+  arrayOfPersons = [];
+
+  constructor(private accessControlLogService: AccessControlLogService) { }
+
+  ngOnInit() {
+    
+  }
+
+  submitForm(form) {
+    this.error = '';
+    this.feedbackEnabled = true;
+    if (form.valid) {
```

Comment:
Here, when the form is valid, is when you push the information to the array calling the accessControlLogService. You can call it twice